src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java

```diff
@@ -27,23 +28,24 @@
 
     private M2FiltersArgumentCollection MTFAC;
     private final double contamination;
+    private final String tumorSample;
 
 
-    public Mutect2FilteringEngine(final M2FiltersArgumentCollection MTFAC) {
+    public Mutect2FilteringEngine(final M2FiltersArgumentCollection MTFAC, final String tumorSample) {
         this.MTFAC = MTFAC;
         contamination = MTFAC.contaminationTable == null ? 0.0 : ContaminationRecord.readContaminationTable(MTFAC.contaminationTable).get(0).getContamination();
-
+        this.tumorSample = tumorSample;
     }
 
     // very naive M1-style contamination filter -- remove calls with AF less than the contamination fraction
     private void applyContaminationFilter(final M2FiltersArgumentCollection MTFAC, final VariantContext vc, final Collection<String> filters) {
-        if (vc.hasAttribute(VCFConstants.ALLELE_FREQUENCY_KEY)) {
-            final double[] alleleFrequencies = getArrayAttribute(vc, VCFConstants.ALLELE_FREQUENCY_KEY);
-            final double maxFrequency = MathUtils.arrayMax(alleleFrequencies);
+        final Genotype tumorGenotype = vc.getGenotypes(tumorSample).get(0);
+            final double[] alleleFractions = GATKProtectedVariantContextUtils.getAttributeAsDoubleArray(tumorGenotype, VCFConstants.ALLELE_FREQUENCY_KEY,
+                    () -> new double[] {1.0}, 1.0);
+            final double maxFrequency = MathUtils.arrayMax(alleleFractions);
```

Comment:
`maxFrequency` -> `maxFraction`. Sorry for being fussy about allele fraction vs frequency but I get confused when they are used interchangeably